### ISSUE TITLE
Default Group

### ISSUE THREAD

**Issue by 0H4rl3qu1n0** [lucko/LuckPerms#1749]
When People Join They are not given the default group for some reason

**Comment by Andre601** [comment]
https://github.com/lucko/LuckPerms/wiki/Default-Groups

**Comment by 0H4rl3qu1n0** [comment]
no info about it lol

**Comment by Andre601** [comment]
what?
The page explains how LP handles default groups and what some possible solutions are...

Also, instead of just saying "When People Join They are not given the default group for some reason" or "no info about it lol" maybe try to actually provide some useful information?

**Comment by LastTulare** [comment]
I am also having the same issue, I am setting up a new server and using my test account to test how new players would interact with the server. Joining the server "for the first time" does not seem to put my account into the default group and also does not create a user file in the storage. I am not seeing anything in https://github.com/lucko/LuckPerms/wiki/Default-Groups that is helping me understand this issue. Using version 4.4.30 on bukkit.

**Comment by Andre601** [comment]
The default groups page already explains that LP doesn't store the player when they don't have any permissions or groups assigned. So that is intentional behaviour.
Do the players have permissions you set in the default group?
If yes, then they have the default group assigned.

**Comment by LastTulare** [comment]
I did see that part. My intention is to start new players into the default group so they can have basic permissions. There is a lot of info to cover and I have been lost on that part. For now, it does not seem like new to server players have any permissions from the default group I have set. I used the migration command successfully just prior to testing.

**Comment by LastTulare** [comment]
I think I figured it out. This is part of the default permissions I got after the migration command. I see some perms have a value under them "value: true" or "value: false". The perms that are missing the value true or false wont work?

- zarena.voter:
    value: true
    world: parkour
- ecocreature.gain.*
- ecocreature.reward.*
- modifyworld.*
- multiverse.access.*
- phatloots.loot.*
- serversigns.use.*
- towny.wild.*
- worldguard.region.teleport.member.*
- backfromthedead.backpack
- backfromthedead.spawn
- bssentials.command.delhome
- bssentials.command.disnick
- bssentials.command.home
- bssentials.command.nick
- bssentials.command.pm
- bssentials.command.sethome
- bssentials.command.spawn
- bssentials.command.viewnick
- bssentials.command.warp
- bukkit.command.plugins:
    value: false
- chatex.allowchat
- chatex.chat.bold
- chatex.chat.color
- chatex.chat.italic
- coloredtags.hero:
    value: false
- coloredtags.hr:
    value: false
- coloredtags.immortal:
    value: false
- coloredtags.initiate:
    value: false
- coloredtags.legend:
    value: false
- coloredtags.nomad
- coloredtags.owner:
    value: false
- coloredtags.vip:
    value: false

**Comment by Turbotailz** [comment]
If the value is undefined then it is assumed to be true.

Could you just post a link to your editor? /LP editor

**Comment by Andre601** [comment]
Please keep in mind that migrating from other perm plugins **won't** change the default group.
The default group is always called "default" and you have to use one of the methods explained on the Default Group page to change that.
My recommendation is to let default inherit the group, that previously was your default one.

Also, it doesn't matter if value: true is set... When it doesn't do LP assumes you're granting the permissions.

**Comment by LastTulare** [comment]
Here is a screenshot of my FTP editor and the link for the LP editor groups

https://luckperms.github.io/editor/#XefH28ZFUF

![LP](https://user-images.githubusercontent.com/4548306/65019690-907e9700-d8e0-11e9-8b55-5537a9f5dbee.png)

**Comment by Andre601** [comment]
We don't encourage people to manually edit files as a small typo can mess up your entire system.

**Comment by LastTulare** [comment]
I have not edited the file yet. that was the result of the migration

**Comment by Turbotailz** [comment]
You have a lot of permissions with world contexts, make sure you are in the correct world while debugging the permissions. You can also use the verbose mode to see whether the permission checks are being passed or not.

**Comment by LastTulare** [comment]
The permissions in question are in the global context, a lot of the perms are dead from my old 2015 server HA! I added a few new ones from BSSENTIALS, and added them to the PEX file before the migration, and they  are near the bottom of the list. At the same time, before I wiped the player file, I was in the group "initiate" and I had the "INITIATE" prefix in chat, but couldnt use basic perms. I deleted the player file, and prefix for default should be "NOMAD" but there is no prefix, and still not able to use basic commands such as /spawn

**Comment by Turbotailz** [comment]
Just reply here or join the discord

**Comment by LastTulare** [comment]
I have it figured out. Deleted the dead perms. Realized some plugin perms are not working because they are either broken or some other issue, like "colored tags" perms are not working due to it being an outdated plugin so I used the prefix perms successfully, which shown me my test account is in the default group displaying the prefix tag I made. spawn is not working because I believe Bssentials permission for spawn is not working. I have it from here. Thank you Turbo and Andre.
